### DISCUSSION TITLE
Self-Hosted edge functions

### DISCUSSION THREAD

**Discussion by Samsonov0** [supabase/supabase#12079] [Questions]
Hi! The team and I had a need to use self-hosted supabase, but we faced a problem in the form of a lack of edge functions. We decided to use deno, as it is done in cloud supabase, but the supabase sdk uses the following url to connect to the project: "https://<project_ref>.supabase.co ". Where can we get the project address we need with self-host?

**Comment by GaryAustin1** [comment]
The supabase-js function call is only for edge functions hosted with a Supabase instance.  So you would need an instance and to use the CLI to deploy them there.

If you created your functions with Deno directly then you would use javascript fetch to access your functions.

**Reply by Samsonov0** [reply]
That is, at the moment there is no way to locally reproduce the behavior of edge functions in the same way as they currently work with a cloud supabase?

**Reply by SteveChurch** [reply]
@Samsonov0 you might find this thread useful. https://github.com/supabase/supabase/discussions/6786

**Comment by laktek** [comment]
@Samsonov0 ~Can you create an issue for making function domain configurable in here https://github.com/supabase/functions-js~

Sorry I mispoke. You should be able to invoke your self-hosted functions by providing a custom `supabaseUrl` when creating the Supabase client. https://supabase.com/docs/reference/javascript/initializing

**Comment by soedirgo** [comment]
As of https://github.com/supabase/supabase/pull/14654, supabase-js should work OOtB with Edge Functions on the [self-hosted Docker setup](https://supabase.com/docs/guides/self-hosting/docker#setting-up-edge-functions). Let us know if you're having issues setting it up.
